crates/db_views/src/report_combined_view.rs

```diff
@@ -0,0 +1,964 @@
+            .eq(false)
+            .or(comment_report::resolved.eq(false))
+            .or(private_message_report::resolved.eq(false)),
+        )
+        // TODO: when a `then_asc` method is added, use it here, make the id sort direction match,
+        // and remove the separate index; unless additional columns are added to this sort
+        .then_desc(ReverseTimestampKey(key::published));
+    } else {
+      query = query.then_desc(key::published);
+    }
+
+    // Tie breaker
+    query = query.then_desc(key::id);
+
+    let res = query.load::<ReportCombinedViewInternal>(conn).await?;
+
+    // Map the query results to the enum
+    let out = res.into_iter().filter_map(map_to_enum).collect();
+
+    Ok(out)
+  }
+}
+
+/// Maps the combined DB row to an enum
+fn map_to_enum(view: ReportCombinedViewInternal) -> Option<ReportCombinedView> {
+  // Use for a short alias
+  let v = view;
+
+  if let (Some(post_report), Some(post), Some(community), Some(unread_comments), Some(counts)) = (
+    v.post_report,
+    v.post.clone(),
+    v.community.clone(),
+    v.post_unread_comments,
+    v.post_counts,
+  ) {
+    Some(ReportCombinedView::Post(PostReportView {
+      post_report,
+      post,
+      community,
+      unread_comments,
+      counts,
+      creator: v.report_creator,
+      post_creator: v.item_creator,
+      creator_banned_from_community: v.item_creator_banned_from_community,
+      creator_is_moderator: v.item_creator_is_moderator,
+      creator_is_admin: v.item_creator_is_admin,
+      creator_blocked: v.item_creator_blocked,
+      subscribed: v.subscribed,
+      saved: v.post_saved,
+      read: v.post_read,
+      hidden: v.post_hidden,
+      my_vote: v.my_post_vote,
+      resolver: v.resolver,
+    }))
+  } else if let (Some(comment_report), Some(comment), Some(counts), Some(post), Some(community)) = (
+    v.comment_report,
+    v.comment,
+    v.comment_counts,
+    v.post.clone(),
+    v.community.clone(),
+  ) {
+    Some(ReportCombinedView::Comment(CommentReportView {
+      comment_report,
+      comment,
+      counts,
+      post,
+      community,
+      creator: v.report_creator,
+      comment_creator: v.item_creator,
+      creator_banned_from_community: v.item_creator_banned_from_community,
+      creator_is_moderator: v.item_creator_is_moderator,
+      creator_is_admin: v.item_creator_is_admin,
+      creator_blocked: v.item_creator_blocked,
+      subscribed: v.subscribed,
+      saved: v.comment_saved,
+      my_vote: v.my_comment_vote,
+      resolver: v.resolver,
+    }))
+  } else if let (Some(private_message_report), Some(private_message)) =
+    (v.private_message_report, v.private_message)
+  {
+    Some(ReportCombinedView::PrivateMessage(
+      PrivateMessageReportView {
+        private_message_report,
+        private_message,
+        creator: v.report_creator,
+        private_message_creator: v.item_creator,
+        resolver: v.resolver,
+      },
+    ))
+  } else {
+    None
+  }
+}
```

Comment:
That seems a bit hacky, since as far as sql is concerned, those columns are nullable. I'd rather keep this conversion simple, and rely on the tests to catch any issues.

Reply:
Silently failing is more hacky and is important to not do in this case. Changing this to a `Queryable` implementation is a simple enough way to fix that, and it will also make deserialization separate from the `load` function. Do it like this:

```
impl<ST> Queryable<ST, Pg> for ReportCombinedView
where
  ReportCombinedViewInternal: Queryable<ST, Pg>,
{
  type Row = ReportCombinedViewInternal::Row;

  fn build(row: Self::Row) -> deserialize::Result<Self> {
    let v = ReportCombinedViewInternal::build(row)?;
    // body of map_to_enum function goes here, with Some and None replaced with Ok and Err(Box::new(UnexpectedNullError))
  }
}
```

Reply:
This should still be done, but not in this PR